### PR TITLE
Feature/fixed models

### PR DESCRIPTION
**Pull request by piotr-major-rmtcfm-com**

### CODE REVIEW

File: app/controllers/Auth/LoginController.js [hunk 1]
```diff
@@ -41,30 +40,27 @@ class LoginController {
...
+        const user = await this.userRepository.getByEmail(email);
 
         if (!user) {
             return response
                 .status(StatusCodes.UNAUTHORIZED)
                 .json({ auth: false, token: null });
         }
 
-        if (Auth.comparePasswords(password, user.getDataValue('password'))) {
+        const userPassword = await this.userRepository.getPassword(user.id);
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/controllers/Auth/LoginController.js]: fetch user only only with password and if passoword is ok then fetch whole user object

File: app/controllers/ToDo/DestroyController.js [hunk 2]
```diff
@@ -17,25 +16,26 @@ class DestroyController {
...
-            todo.destroy()
-                .then(() => {
-                    response.sendStatus(StatusCodes.NO_CONTENT);
-                })
-                .catch(next);
-        });
+        const todo = await this.todoRepository.findById(todoId);
+
+        if (!todo) {
+            return response.sendStatus(StatusCodes.NOT_FOUND);
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/controllers/ToDo/DestroyController.js]: 204 in destroy controller

File: app/controllers/ToDo/IndexController.js [hunk 3]
```diff
@@ -44,13 +40,16 @@ class IndexController {
...
-        Todo.findAll({ where: { userId: request.loggedUserId } })
-            .then(todos => {
-                return response.json({ todos });
-            })
-            .catch(next);
+        const todos = await this.todoRepository.findAll({
+            where: { userId: request.loggedUserId }
+        });
+
+        return response.json({ todos });
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/controllers/ToDo/IndexController.js]: response.sedn(todos)

File: app/controllers/ToDo/PatchController.js [hunk 4]
```diff
@@ -24,25 +22,27 @@ class PatchController {
      *  @apiError BadRequest    The <code>id</code> of the ToDo element was not found, <code>id</code> does not exist in table ToDo and parameter "name" is not specified
      *  @apiError Forbidden     ToDo element belongs to other User
      */
+    constructor(todoRepository) {
+        this.todoRepository = todoRepository;
+    }
+
     async invoke(request, response, next) {
         const todoId = request.params.id;
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/controllers/ToDo/PatchController.js]: const {id, body:fields ....

File: app/controllers/ToDo/PatchController.js [hunk 5]
```diff
@@ -24,25 +22,27 @@ class PatchController {
...
+            return response.sendStatus(StatusCodes.NOT_FOUND);
+        }
+
+        if (todo.userId !== request.loggedUserId) {
+            return response.sendStatus(StatusCodes.FORBIDDEN);
+        }
+
+        await todo.update(fields, { fields: ['name', 'userId', 'completed'] });
+
+        return response.sendStatus(StatusCodes.OK);
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/controllers/ToDo/PatchController.js]: return updated object

File: app/controllers/ToDo/ShowController.js [hunk 6]
```diff
@@ -34,18 +32,20 @@ class ShowController {
...
+        const todo = await this.todoRepository.findById(todoId);
+
+        if (!todo) {
+            return response.sendStatus(StatusCodes.NOT_FOUND);
+        }
 
-                return response.json(todo);
-            })
-            .catch(next);
+        return response.json(todo);
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/controllers/ToDo/ShowController.js]: send

File: app/controllers/ToDo/ShowController.js [hunk 7]
```diff
@@ -34,18 +32,20 @@ class ShowController {
...
-                if (!todo) {
-                    return response.sendStatus(StatusCodes.NOT_FOUND);
-                }
+        const todo = await this.todoRepository.findById(todoId);
+
+        if (!todo) {
+            return response.sendStatus(StatusCodes.NOT_FOUND);
+        }
 
-                return response.json(todo);
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/controllers/ToDo/ShowController.js:46]: shouldnt be there check if todo belongs to logged user?

File: app/controllers/ToDo/StoreController.js [hunk 8]
```diff
@@ -56,11 +57,9 @@ class StoreController {
             data.userId = request.loggedUserId;
         }
 
-        Todo.create({ ...data })
-            .then(todo => {
-                return response.status(StatusCodes.CREATED).json(todo);
-            })
-            .catch(next);
+        const todo = await this.todoRepository.create({ ...data });
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/controllers/ToDo/StoreController.js]: create(data)

File: app/controllers/ToDo/StoreController.js [hunk 9]
```diff
@@ -56,11 +57,9 @@ class StoreController {
             data.userId = request.loggedUserId;
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/controllers/ToDo/StoreController.js:58]: empty line before `if`

File: app/controllers/ToDo/StoreController.js [hunk 10]
```diff
@@ -47,6 +45,9 @@ class StoreController {
      *        ]
      *    }
      */
+    constructor(todoRepository) {
+        this.todoRepository = todoRepository;
+    }
 
     async invoke(request, response, next) {
         const data = request.body;
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/controllers/ToDo/StoreController.js]: destructructure it

File: app/controllers/ToDo/UpdateController.js [hunk 11]
```diff
@@ -41,35 +39,38 @@ class UpdateController {
      *            ]
      *    }
      */
-    async invoke(request, response, next) {
+    constructor(todoRepository) {
+        this.todoRepository = todoRepository;
+    }
+
+    async invoke(request, response) {
         const todoId = request.params.id;
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/controllers/ToDo/UpdateController.js]: destructure

File: app/controllers/ToDo/UpdateController.js [hunk 12]
```diff
@@ -41,35 +39,38 @@ class UpdateController {
...
-            if (!todo) {
-                return Todo.create({ ...fields, todoId })
-                    .then(todo => {
-                        return response.status(StatusCodes.CREATED).json(todo);
-                    })
-                    .catch(next);
-            }
+        const todo = await this.todoRepository.findById(todoId);
+
+        if (!todo) {
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/controllers/ToDo/UpdateController.js:60]: if there is no todo then just send 404

File: app/controllers/ToDo/UpdateController.js [hunk 13]
```diff
@@ -41,35 +39,38 @@ class UpdateController {
...
+        await todo.update(fields, { fields: ['name', 'userId'] });
 
-            return todo
-                .update(fields, { fields: ['name', 'userId'] })
-                .then(() => {
-                    return response.sendStatus(StatusCodes.OK);
-                })
-                .catch(next);
-        });
+        return response.sendStatus(StatusCodes.OK);
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/controllers/ToDo/UpdateController.js]: return updated object

File: app/controllers/User/DestroyController.js [hunk 14]
```diff
@@ -18,25 +16,24 @@ class DestroyController {
...
+        this.userRepository = userRepository;
+    }
+
+    async invoke(request, response) {
+        const user = await this.userRepository.findById(request.params.id);
 
-    async invoke(request, response, next) {
-        const userId = request.params.id;
+        if (!user) {
+            return response.sendStatus(StatusCodes.NOT_FOUND);
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/controllers/User/DestroyController.js]: 204 in delete

File: app/controllers/User/IndexController.js [hunk 15]
```diff
@@ -33,12 +31,14 @@ class IndexController {
...
-            })
-            .catch(next);
+    constructor(userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    async invoke(request, response) {
+        const users = await this.userRepository.findAll();
+
+        return response.json({ users });
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/controllers/User/IndexController.js]: send(users)

File: app/controllers/User/ShowController.js [hunk 16]
```diff
@@ -16,18 +15,18 @@ class ShowController {
...
-                    return response.sendStatus(StatusCodes.NOT_FOUND);
-                }
+        if (!user) {
+            return response.sendStatus(StatusCodes.NOT_FOUND);
+        }
 
-                response.json(user);
-            })
-            .catch(next);
+        return response.json(user);
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/controllers/User/ShowController.js]: send

File: app/controllers/User/StoreController.js [hunk 17]
```diff
@@ -46,17 +46,19 @@ class StoreController {
...
      */
+    constructor(userRepository) {
+        this.userRepository = userRepository;
+    }
+
     async invoke(request, response, next) {
-        await User.create({ ...request.body })
-            .then(user => {
-                const token = Auth.signIn(user);
+        const user = await this.userRepository.create({ ...request.body });
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/controllers/User/StoreController.js]: create(request.body)

File: app/controllers/User/StoreController.js [hunk 18]
```diff
@@ -46,17 +46,19 @@ class StoreController {
...
+
+        const token = await Auth.signIn(user);
 
-                let plainUser = user.get({ plain: true });
-                delete plainUser.password;
+        let plainUser = user.get({ plain: true });
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/controllers/User/StoreController.js]: refetch user without password, I mean exclude password in defaultScope

File: app/controllers/User/UpdateController.js [hunk 19]
```diff
@@ -50,25 +48,27 @@ class UpdateController {
...
-            }
+        await user.update(fields);
 
-            user.update(fields)
-                .then(() => {
-                    response.sendStatus(StatusCodes.OK);
-                })
-                .catch(next);
-        });
+        return response.sendStatus(StatusCodes.OK);
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/controllers/User/UpdateController.js]: return updated object

File: app/routes/reset-password.js [hunk 20]
```diff
@@ -1,11 +1,21 @@
 const express = require('express');
 const router = express.Router();
 
-const ResetPasswordController = require('../controllers/ResetPasswordController');
-const changePasswordRequest = require('../requests/ChangePasswordRequest');
-const ResetPasswordRequest = require('../requests/ResetPasswordRequest');
+module.exports = di => {
+    const ResetPasswordController = require('../controllers/ResetPasswordController');
+    const changePasswordRequest = require('../requests/ChangePasswordRequest');
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/routes/reset-password.js]: why does it start with lowerletter?

File: app/validators/todoValidator.js [hunk 21]
```diff
@@ -0,0 +1,17 @@
+const { check } = require('express-validator');
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/validators/todoValidator.js]: use `body`

File: app/validators/userValidator.js [hunk 22]
```diff
@@ -0,0 +1,112 @@
+const { check } = require('express-validator');
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/validators/userValidator.js]: use `body`

File: app/routes/reset-password.js (hunk 20 above)
Comment (piotr-major-rmtcfm-com) [reply app/routes/reset-password.js]: it was like this

Comment (daniel-pupczyk-rmtcfm-com) [reply app/routes/reset-password.js]: please fix it

File: app/controllers/ToDo/ShowController.js (hunk 7 above)
Comment (piotr-major-rmtcfm-com) [reply app/controllers/ToDo/ShowController.js:46]: I think it should. I will change this everywhere , in next pr

File: app/controllers/ToDo/UpdateController.js (hunk 12 above)
Comment (piotr-major-rmtcfm-com) [reply app/controllers/ToDo/UpdateController.js:60]: I think it would be better to improve in next pr - this one + new tests for this and patchController

**Pull request merged**